src/main/java/io/confluent/copycat/hdfs/WALFactory.java

```diff
@@ -0,0 +1,28 @@
+/**
+ * Copyright 2015 Confluent Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License
+ * is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
+ * or implied. See the License for the specific language governing permissions and limitations under
+ * the License.
+ **/
+
+package io.confluent.copycat.hdfs;
+
+import org.apache.kafka.common.TopicPartition;
+
+public class WALFactory {
+
+  public static WAL createWAL(String topicsDir, TopicPartition topicPart, Storage storage) {
+    if (storage instanceof HdfsStorage) {
```

Comment:
Could you make this a method of `Storage` so you don't need this class? i.e. just 
```
WAL wal(String topicsDir, TopicPartition topicPart);
```
in the interface and 
```
WAL wal(String topicsDir, TopicPartition topicPart) {
   return new MemoryWAL(topicsDir, topicPart, this);
}
```
in `MemoryStorage`.